api/rooms/views/map.py

```diff
@@ -0,0 +1,113 @@
+import json
+from pyproj            import Transformer
+
+from django.db.models  import Q
+from django.views      import View
+from django.http       import JsonResponse
+
+from ..models          import Room, TradeType
+
+def parse_wgs84_to_utmk(longitude, latitude):
+    return Transformer.from_crs("epsg:4326", "epsg:5178").transform(latitude, longitude)
+
+def parse_utmk_to_wgs84(utmk_x, utmk_y):
+    return Transformer.from_crs("epsg:5178", "epsg:4326").transform(utmk_x, utmk_y)
+
+def room_filter(func):
+    def wrapper(self, request, *args, **kwargs):
+        filters = json.loads(request.GET.get('filter', '{}'))
+        room_types  = filters.get('roomType', [])
+        trade_types = filters.get('tradeType', [])
+        locations  = json.loads(request.GET.get('location', '[]'))
+        dong_code  = request.GET.get('dongCode')
+        q          = Q()
+
+        if len(locations) == 2 and \
+            len(locations[0]) == 2 and len(locations[1]) == 2:
+            sw = parse_wgs84_to_utmk(locations[0][0], locations[0][1])
+            ne = parse_wgs84_to_utmk(locations[1][0], locations[1][1])
+
+            q &= Q(location__utmk_x__range = [sw[0], ne[0]], 
+                location__utmk_y__range = [sw[1], ne[1]])
+
+        if trade_types:
+            trade_q      = Q()
+            deposit      = filters.get('deposit', [0, 999999999])
+            monthly_rent = filters.get('monthlyRent', [0, 999999999])
+            sale         = filters.get('sale', [0, 999999999])
+            
+            if TradeType.Enum.MONTHLY_RENT.value in trade_types :
+                trade_q |= Q(
+                    trade_type_id=TradeType.Enum.MONTHLY_RENT.value, 
+                    monthly_rent__range=monthly_rent, deposit__range=deposit)
+
+            if TradeType.Enum.DEPOSIT.value in trade_types :
+                trade_q |= Q(trade_type_id=TradeType.Enum.DEPOSIT.value, 
+                    deposit__range=deposit)
+
+            if TradeType.Enum.SALE.value in trade_types :
+                trade_q |= Q(trade_type_id=TradeType.Enum.SALE.value,
+                    sale__range=sale)
+
+            q &= Q(trade_q)
+
+        if room_types:
+            q &= Q(room_type_id__in=room_types)
+
+        if dong_code:
+            q &= Q(location__dong_code=dong_code)
+
+        request.q = q
+        
+        return func(self,request, *args, **kwargs)
+
+    return wrapper
+
+class RoomListView(View):
+    @room_filter
+    def get(self, request): 
+        next_page = 2
+        has_more  = False
+        offset    = int(request.GET.get('offset', 0))
+        limit     = int(request.GET.get('limit', 15))
+
+        rooms = Room.objects \
+            .select_related('location', 'room_type', 'trade_type') \
+            .prefetch_related('roomimage_set').filter(request.q)
```

Comment:
조회할때부터 `offset`, `limit`을 붙여주면 필요한 양만 불러올 수 있을 것 같은데 어떠신가요?
아래 has_more이 꼭 필요한 코드가 아니라면 line 78~82 코드가 없어도 될 것 같아요!

```suggestion
        rooms = Room.objects \
            .select_related('location', 'room_type', 'trade_type') \
            .prefetch_related('roomimage_set').filter(request.q)[offset:offset+limit]
```

Reply:
1. 검색 조건에 해당되는 방의 총 갯수를 화면에 찍어야 되기때문에, 페지네이션 전에 전체 data의 갯수를 알아야 되서 분리하였습니다.
 - 혹시 이러한 사항때문에 분리하였는데, 다른 좋은 방법이 있으면 추천해주시면 감사하겠습니다.

Reply:
그렇다고 하면 이경우에는 room 전체를 가져올 때, `location`, `room_type`, `trade_type` ... 전체를 같이 가져오기 때문에, 
offset, limit으로 가져오는 수를 제한하고, Room 전체 count만하는 쿼리를 추가로 날리는게 좋을 것 같아요!

Reply:
```python
total_count = Room.objects.all().count()
```

Reply:
태우님 생각은 어떠신가요?

Reply:
이 경우에는 rooms 테이블에 있는 모든 데이터의 갯수를 카운팅 하기때문에 제가 생각한 의도와 다르게 동작할것 같습니다.

저는 필터링 조건에 만족하는 데이터의 전체 갯수가 몇개인지 알고 싶어서 필터 후의 카운트를 구했습니다.